src/linux/mod.rs

```diff
@@ -99,10 +99,7 @@ fn restrict_networking() -> Result<()> {
     let result = create_user_namespace(true).and_then(|_| unshare(Namespaces::NETWORK));
 
     // Apply seccomp network filter.
-    match NetworkFilter::apply() {
-        _ if result.is_ok() => Ok(()),
-        result => result,
-    }
+    result.or_else(|_| NetworkFilter::apply())
```

Comment:
Wait... Now the seccomp filter is only applied if the namespace failed... Is that intended?